sqlglot/dialects/doris.py

```diff
@@ -22,6 +22,48 @@ def _lag_lead_sql(self, expression: exp.Lag | exp.Lead) -> str:
     )
 
 
+# Accept both DATE_TRUNC(datetime, unit) and DATE_TRUNC(unit, datetime)
+def _build_date_trunc(args: list[exp.Expression]) -> exp.Expression:
+    a0, a1 = seq_get(args, 0), seq_get(args, 1)
+
+    def _is_unit_like(e: exp.Expression | None) -> bool:
```

Comment:
I'm not sure I like this approach. Perhaps a simpler heuristic could be: if `e` is not a string literal, then it can't be a unit (I assume?). Otherwise, `e` is a string, so if there's at least one digit in it, it also can't be a unit. Otherwise, it's a unit.